src/StoreApi/Routes/Checkout.php

```diff
@@ -310,6 +310,18 @@ private function create_or_update_draft_order( \WP_REST_Request $request ) {
 			$this->order_controller->update_order_from_cart( $this->order );
 		}
 
+		// Confirm order is valid before proceeding further.
```

Comment:
Handling for no order should be done by the `woocommerce_blocks_checkout_update_order_meta`. Swapping these conditionals around could mean that someone replaces the order object via `woocommerce_blocks_checkout_update_order_meta` leaving the API in an error state.

Reply:
>  could mean that someone replaces the order object

Can you share an example? Could a callback replace the object passed into it as an argument and have that change persist (passing by reference?).

If yes, then I see your point here!

Reply:
Yeah I believe these objects are passed by reference, so this guard is in place to prevent the route trying to format an invalid order object.